lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * API to interact with global settings & styles.
+ *
+ * @package gutenberg
+ */
+
+if ( ! function_exists( 'wp_theme_has_theme_json' ) ) {
+	/**
+	 * Whether a theme or its parent have a theme.json file.
+	 *
+	 * @param boolean $use_cache Whether data should be computed from scratch or use the cache, if available
+	 *
+	 * @return boolean
+	 */
+	function wp_theme_has_theme_json( $use_cache = true ) {
+		static $theme_has_support = null;
+
+		if ( false === $use_cache ) {
+			$theme_has_support = null;
+		}
+
+		if ( isset( $theme_has_support ) ) {
```

Comment:
nitpick: Using a strict `null !==` check would be easier to understand
```suggestion
		if ( null !== $theme_has_support ) {
```